File: packages/swift/Sources/CatalystSwift/Models/Token.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

public struct Token: Codable, Sendable {
  public let accessToken: String
  public let refreshToken: String
  public let tokenType: String
  public let scope: String?

  /// ISO8601 string, or nil when unset. Kept as a raw string (rather than `Date`) since the
  /// OAuth token-exchange decoder does not configure a date decoding strategy.
  public let expiresAt: String?
  public let expiresIn: Int?
}

File: packages/nodejs/src/types/token.ts
export interface Token {
  accessToken: string;
  refreshToken: string;
  tokenType: string;
  /** Present on responses from the real token-exchange endpoint (e.g. "read write"). */
  scope?: string;
  /** ISO8601 timestamp string for when the access token expires. */
  expiresAt?: string;
  /** Seconds until the access token expires. */
  expiresIn?: number;
}
